account_asset_management/migrations/12.0.1.0.0/post-migration.py

```diff
@@ -18,20 +18,44 @@ def adjust_asset_values(env):
         FROm account_asset_profile aap
         WHERE aa.profile_id = aap.id""",
     )
-    # Adjust method_time, method_number and method_period
+    # Adjust method_time, method_end, method_number and method_period
     number = sql.Identifier(openupgrade.get_legacy_name('method_number'))
     period = sql.Identifier(openupgrade.get_legacy_name('method_period'))
+    time = sql.Identifier(openupgrade.get_legacy_name('method_time'))
     for table in ['account_asset_profile', 'account_asset']:
         table = sql.Identifier(table)
         openupgrade.logged_query(
             env.cr, sql.SQL("""
             UPDATE {table}
             SET method_time = 'year',
+                method_end = NULL,
                 method_number = ({number} * {period}) / 12
-            WHERE MOD({number} * {period}, 12) = 0
+            WHERE MOD({number} * {period}, 12) = 0 AND {time} != 'end'
             """).format(
                 number=number,
                 period=period,
+                time=time,
+                table=table,
+            ),
+        )
+        openupgrade.logged_query(
+            env.cr, sql.SQL("""
+        UPDATE {table}
+        SET method_time = 'year',
+            method_number = 0
+        WHERE {time} = 'end'
+        """).format(
+                time=time,
```

Comment:
Incorrect indentation

Reply:
Is this needed anyway?

Reply:
Obviously is needed. If you don't convert the method_time 'end' to 'year', then after migration you have some records with this non existing method.

Reply:
And the indentation is correct.

Reply:
Yeah, but that was the previous situation, for not destructing data.

Reply:
I don't see indentation as correct.

Reply:
Pycharm says me it's correct.

Reply:
Indentation depends in where the string ends. I changed it, to make it homogeneous with the other queries, but the indentation was still correct before.

Reply:
Yes, all linters are tolerant to more indentation than the minimal (you can put 40 spaces and it will be correct). The only lint is for incorrect deindentations, but that doesn't mean it's strictly "correct", as it reduces the readability.